includes/gateway/class-omise-payment.php

```diff
@@ -226,6 +226,62 @@ public function process_payment( $order_id ) {
 	abstract public function result( $order_id, $order, $charge );
 
 	/**
+	 * Process refund.
+	 *
+	 * @param  int    $order_id
+	 * @param  float  $amount
+	 * @param  string $reason
+	 *
+	 * @return boolean True|False based on success, or a WP_Error object.
+	 *
+	 * @see    WC_Payment_Gateway::process_refund( $order_id, $amount = null, $reason = '' )
+	 */
+	public function process_refund( $order_id, $amount = null, $reason = '' ) {
+		if ( ! $order = wc_get_order( $order_id ) ) {
+			$message = __(
+				'Refund failed. Cannot retrieve an order from the given ID: %s. Please try again or you may do a manual refund.', 
+				'omise'
+			);
+
+			return new WP_Error( 'error', sprintf( wp_kses( $message, array( 'br' => array() ) ), $order_id ) );
+		}
+
+		try {
+			$charge = OmiseCharge::retrieve( $order->get_transaction_id() );
+			$refund = $charge->refunds()->create( array(
+				'amount' => Omise_Money::to_subunit( $amount, $order->get_order_currency() )
```

Comment:
`to_subunits`?

Reply:
Noted. This one requires some change that does not related to this PR.
Probably another PR will be better if we want to change this. (but very low priority)